server/channels/api4/system.go

```diff
@@ -401,6 +403,42 @@ func getLogs(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(model.ArrayToJSON(lines)))
 }
 
+func downloadLogs(c *Context, w http.ResponseWriter, r *http.Request) {
+
+	lines, _ := c.App.GetLogs(c.AppContext, c.Params.Page, c.Params.LogsPerPage)
+
+	tmpfile, err := os.CreateTemp("", "logs-*.log")
```

Comment:
Instead of using a temp file for this, you can put all of this into a `ByteReader` to avoid writing to and then reading from disk. `ByteReader` also implements the `io.ReadCloser` interface that `WriteFileResponse` needs. I think that'll look like this:

```go
	lines, _ := c.App.GetLogs(c.AppContext, c.Params.Page, c.Params.LogsPerPage)
	b := bytes.NewReader([]byte(model.ArrayToJSON(lines)))

	web.WriteFileResponse("mattermost.log",
		"text/plain",
		int64(b.Len()),
		time.Now(),
		*c.App.Config().ServiceSettings.WebserverMode,
		b,
		true,
		w,
		r)
```

Reply:
Alternatively, you could probably open the actual log file and serve that. I tried that, and while it was much quicker on my local instance, it also tried downloading my entire 70MB log file, and I don't know if we want to do that